synopsys-task/src/synopsys-task/validator.ts

```diff
@@ -37,6 +38,8 @@ export function validateParameters(
   if (invalidParams.length > 0) {
     errors.push(
       `[${invalidParams.join()}] - required parameters for ${toolName} is missing`
+        .concat(constants.SPACE)
+        .concat(ErrorCode.MISSING_REQUIRED_PARAMETERS.toString())
```

Comment:
what will be error message in ADO related errors?

Reply:
I concatenated the respective error code to the existing exception message as we are not logging any exception message here. In the `main.ts` file, exceptions are caught and the associated error message will be displayed from `EXIT_CODE_MAP`.

For example in the case of an unsuccessful BlackDuck execution, the error message will be like this:
`2024-05-06T11:31:12.6180833Z ##[error]Workflow failed! Exit Code: 2 Error from
adapter end`

Reply:
I want to know the error messages related to ADO validations i.e. if blackduck token is missing..?
will it be same or there is any change?

Reply:
If Blackduck, Coverity, or Polaris mandatory parameters are missing, an error message will be shown with the exit code appended at the end.